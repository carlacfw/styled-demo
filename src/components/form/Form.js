import React, { Component } from 'react'
import styled from 'styled-components'
import media from '../../util/media'

const FormWapper = styled.form`
  display: flex;
  flex-direction: column;
  margin: 3em;
`

const Input = styled.input`
  color: black;
  margin: 0 10px;
  float: left;
`

const ResponsiveWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  margin: 20px;
  justify-content: space-between;
  ${media.desktop`
    flex-direction: row;
  `};
`

const SubmitInput = styled.input`
  background-color: darkblue;
  border-radius: 10px;
  color: white;
  width: 150px;
  height: 30px;
  margin: 10px;
  &:disabled {
    background: lightgray;
  }
`

const Label = styled.label`
  margin: 10px;
`

export default class extends Component {
  constructor(props) {
    super(props)
    this.state = {
      loading: false,
      name: '',
      hair: '',
      eye: ''
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange(change) {
    this.setState(change)
  }

  handleSubmit(event) {
    event.preventDefault()
    this.setState({ loading: true })
  }

  render() {
    return (
      <FormWapper onSubmit={this.handleSubmit}>
        <ResponsiveWrapper>
          <Label htmlFor="name">Name</Label>
          <Input
            id="name"
            onChange={event => this.handleChange({ name: event.target.value })}
          />
        </ResponsiveWrapper>
        <ResponsiveWrapper>
          <Label htmlFor="hair">Hair colour</Label>
          <Input
            id="hair"
            onChange={event => this.handleChange({ hair: event.target.value })}
          />
        </ResponsiveWrapper>
        <ResponsiveWrapper>
          <Label htmlFor="eye">Eye Colour</Label>
          <Input
            id="eye"
            onChange={event => this.handleChange({ eye: event.target.value })}
          />
        </ResponsiveWrapper>
        <SubmitInput type="submit" disabled={this.state.loading} />
      </FormWapper>
    )
  }
}
