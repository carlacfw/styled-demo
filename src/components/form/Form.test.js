import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Form from './Form'
describe('The Form ', () => {
  it('It renders the Form correctly', () => {
    const wrapper = shallow(<Form />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
