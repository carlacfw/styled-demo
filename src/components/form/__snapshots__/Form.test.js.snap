// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Form  It renders the Form correctly 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.h4>
     Person form 
  </styled.h4>
  <styled.div>
    <styled.label
      htmlFor="name"
    >
      Name
    </styled.label>
    <styled.input
      id="name"
      onChange={[Function]}
    />
  </styled.div>
  <styled.div>
    <styled.label
      htmlFor="hair"
    >
      Hair colour
    </styled.label>
    <styled.input
      id="hair"
      onChange={[Function]}
    />
  </styled.div>
  <styled.div>
    <styled.label
      htmlFor="eye"
    >
      Eye Colour
    </styled.label>
    <styled.input
      id="eye"
      onChange={[Function]}
    />
  </styled.div>
  <styled.input
    disabled={false}
    type="submit"
  />
</styled.form>
`;
